python/examples/llama/test_llama3.py

```diff
@@ -0,0 +1,1889 @@
+from dataclasses import dataclass
+import math as pymath
+import pytest
+import torch
+from typing import Optional, Tuple
+
+from mlir import ir
+from mlir.dialects import transform, func, linalg, tensor, arith, complex, math
+from mlir.dialects.transform import structured
+from mlir.dialects.transform import interpreter
+from mlir.passmanager import PassManager
+from mlir.runtime.np_to_memref import (
+    get_ranked_memref_descriptor,
+)
+from mlir.execution_engine import ExecutionEngine
+
+
+from lighthouse import utils as lh_utils
+
+
+@dataclass
+class ModelArgs:
+    dim: int = 4096
+    n_layers: int = 32
+    n_heads: int = 32
+    n_kv_heads: Optional[int] = None
+    vocab_size: int = -1
+    multiple_of: int = 256  # make SwiGLU hidden layer size multiple of large power of 2
+    ffn_dim_multiplier: Optional[float] = None
+    norm_eps: float = 1e-5
+    rope_theta: float = 500000
+
+    max_batch_size: int = 32
+    max_seq_len: int = 2048
+
+
+def affine_map(dim_count, exprs, *, symb_count=0):
+    return ir.AffineMap.get(dim_count, symb_count, exprs)
+
+
+parallel = linalg.IteratorType.parallel
+reduction = linalg.IteratorType.reduction
+
+
+def create_pass_pipeline(ctx: ir.Context) -> PassManager:
+    with ctx:
+        pm = PassManager("builtin.module")
```

Comment:
```suggestion
def create_pass_pipeline() -> PassManager:
    pm = PassManager("builtin.module")
```

See comment below for how to elide dealing with contexts in most places.

Reply:
See https://github.com/libxsmm/tpp-mlir/blob/37a498bd1e320e00fa50e3323cbaac2867cd7a1e/python/mlir/tpp/sched/bundles.py#L41-L43 for an example for dealing with passes that expect to run on particular ops w.r.t to the root module.